backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/purchasecontract/averageonshippedqty/AverageAVOnShippedQtyComputingMethod.java

```diff
@@ -59,14 +58,13 @@
 
 @Component
 @RequiredArgsConstructor
-public class AverageAVOnShippedQtyComputingMethod implements IComputingMethodHandler
+public class AverageAVOnShippedQtyComputingMethod extends AbstractComputingMethodHandler
 {
-	@NonNull
-	private final IInOutDAO inOutDAO = Services.get(IInOutDAO.class);
-	@NonNull
-	private final ModularContractProvider contractProvider;
-	@NonNull
-	private final ComputingMethodService computingMethodService;
+	@NonNull private final IInOutDAO inOutDAO = Services.get(IInOutDAO.class);
+	@NonNull private final IProductBL productBL = Services.get(IProductBL.class);
+
+	@NonNull private final ModularContractProvider contractProvider;
+	@NonNull private final ComputingMethodService computingMethodService;
 
 	private final CurrencyPrecision precision = CurrencyPrecision.ofInt(4);
```

Comment:
pls use standard precission of currency here

Reply:
I would not change the functionality since I don't know the reason why 4 was used. Please, contact @cristinamghita

Reply:
https://github.com/metasfresh/metasfresh/pull/18253